core/tx_pool.go

```diff
@@ -901,14 +934,15 @@ func (pool *TxPool) addTxsLocked(txs []*types.Transaction, local bool) []error {
 	erroredTxns := []types.RPCTransactionError{}
 
 	for i, tx := range txs {
-		var replace bool
-		if replace, errs[i] = pool.add(tx, local); errs[i] == nil && !replace {
+		replace, err := pool.add(tx, local)
+		if err == nil && !replace {
 			from, _ := types.Sender(pool.signer, tx) // already validated
 			dirty[from] = struct{}{}
 		}
-		if errs[i] != nil && errors.Cause(errs[i]) != ErrKnownTransaction {
-			erroredTxns = append(erroredTxns, *types.NewRPCTransactionError(tx.Hash(), errs[i]))
+		if err != nil && err != ErrKnownTransaction {
+			erroredTxns = append(erroredTxns, *types.NewRPCTransactionError(tx.Hash(), err))
 		}
+		errs[i] = errors.Cause(err)
```

Comment:
why were these lines here changed? Not sure what the change adds

Reply:
The function should return error causes and not wrapped errors with messages. Look at `validateTx`.